BangazonAPI/BangazonAPI/Controllers/ProductController.cs

```diff
@@ -1,46 +1,262 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Data.SqlClient;
 using System.Linq;
 using System.Threading.Tasks;
+using BangazonAPI.Models;
 using Microsoft.AspNetCore.Http;
 using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Configuration;
 
 namespace BangazonAPI.Controllers
 {
     [Route("api/[controller]")]
     [ApiController]
     public class ProductController : ControllerBase
     {
+         
+        private readonly IConfiguration _config;
+
+        public ProductController(IConfiguration config)
+        {
+            _config = config;
+        }
+
+        public SqlConnection Connection
+        {
+            get
+            {
+                return new SqlConnection(_config.GetConnectionString("DefaultConnection"));
+            }
+        }
         // GET: api/Product
         [HttpGet]
-        public IEnumerable<string> Get()
+        public async Task<IActionResult> Get()
         {
-            return new string[] { "value1", "value2" };
+            using (SqlConnection conn = Connection)
+            {
+                conn.Open();
+                using (SqlCommand cmd = conn.CreateCommand())
+                {
+                    cmd.CommandText = "SELECT Id, Price, Title, Description, Quantity, IsArchived, ProductTypeId, CustomerId FROM Product";
+                    SqlDataReader reader = cmd.ExecuteReader();
+                    List<Product> products = new List<Product>();
+
+                    while (reader.Read())
+                    {
+                        
+                        Product product = new Product
+                        {
+                            Id = reader.GetInt32(reader.GetOrdinal("Id")),
+                            Price = reader.GetInt32(reader.GetOrdinal("Price")),
```

Comment:
I got an 500 error (and failing tests) for GET one and GET all. In the SQL script you sent me, the `price` property is of type money, which is a decimal. You'd need to use `reader.GetDecimal()` instead of `reader.GetInt32()`. Did this throw errors for anyone else?